### PR TITLE
docs: Adding /doc folder and workspace troubleshooting files.

### PR DESCRIPTION
**Pull request by travberk**
No Issue number.

Description of changes:  added a /docs folder and included troubleshooting information for workspace lifecycle events.

Checklist:
These changes do not affect the code base.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by bhudnell** [comment]
Hi @travberk. It looks like this documentation is Service Workbench specific? If that is the case, I think this documentation is better suited for the `solutions/swb-reference` folder.

**Comment by travberk** [comment]
> Hi @travberk. It looks like this documentation is Service Workbench specific? If that is the case, I think this documentation is better suited for the `solutions/swb-reference` folder.

Yes, it is SWB specific.  I put it there at the request of @nguyen102.  I can move it where you ask if that's what the team wants.

File: docs/troubleshooting-lifecycle-events.md [hunk 1]
```diff
@@ -0,0 +1,32 @@
+# Troubleshooting workspace lifecycle events #
+
+## Launching/Terminating a Workspace ##
+
+![Diagram 1 - worfklow for Launching/Terminating a workspace](./launch-terminate-workflow.jpg)
+
+A user will initiate the request through the UI, which in turn makes an API request.  The API Gaateway calls the "apiHandler" Lambda function.  Each execution will execute a Systems Manager Automation within the hosting account.  It will also make a new entry in the DynamoDB table named ``swb-<stage>-<awsRegionShortName>`` For example, the table might be named ``swb-dev-va``.  (See item 1 in the diagram.)
```
Comment (nguyen102) [inline docs/troubleshooting-lifecycle-events.md:7]: `Gaateway` => `Gateway`.
```
``swb-<stage>-<awsRegionShortName>`` => Only one ` is enough, you don't need two of them.
```

File: docs/troubleshooting-lifecycle-events.md [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+The "statusHandler" Lambda function is also responsible for updating the DynamoDB table entries for the workspace.  (See item 1 in the diagram.)  The DynamoDB values are reflected in the UI as the workspace's status for the user to see.
+
+
+## Starting/Stopping a Workspace ##
+
+![Diagram 2 - worfklow for Starting/Stopping a workspace](./start-stop-workflow.jpg)
+
+A user will initiate the request through the UI, which will make an API request.  In turn, the Lambda function "apiLambda" will be called.  Each execution will generate a CloudWatch log in the group name that begins with ``/aws/lambda/swb-<stage>-<awsRegionShortName>-apiLambda``.  For example, the log group may be named ``/aws/lambda/swb-dev-va-apiHandler``.  (See item 1 in the diagram.)  There will be a log entry to idenfity the role that the Lambda function assumes in the Hosting account.  The entry will look like this:
+
+	2022-08-17T20:20:24.216Z	884d2b1a-a1b2-c3d4-e5f6-8b981c764c22	INFO	Assuming EnvMgmt role arn:aws:iam::123456789012:role/swb-dev-va-env-mgmt with externalId workbench
```
Comment (nguyen102) [inline docs/troubleshooting-lifecycle-events.md:26]: We might delete this log in the future, so I wouldn't put that here. However, any errors will be shown in CWL at the log group u specified.

**Review by nguyen102 — COMMENTED** [review]
Thanks for adding this Travis. I added a couple of comments.

**Comment by bhudnell** [comment]
> > Hi @travberk. It looks like this documentation is Service Workbench specific? If that is the case, I think this documentation is better suited for the `solutions/swb-reference` folder.
> 
> Yes, it is SWB specific. I put it there at the request of @nguyen102. I can move it where you ask if that's what the team wants.

@nguyen102 thought on what folder the docs should reside in?

**Review by bhudnell — DISMISSED** [review]

File: solutions/swb-reference/README.md [hunk 3]
```diff
@@ -111,6 +111,10 @@ Go to `solutions/swb-app` to update `staticRouteConfig.ts` and `staticPermission
 3. Run `STAGE=<STAGE> node generateCognitoTokens.js <userName> '<password>'` with the correct value for `<userName>` and `<password>`. It should be a user that has been created for your SWB deployment. Note, the quotes around `<password>` is necessary for the script to correctly parse passwords that have symbols in it. 
 4. In the console output, use the `accessToken` that is provided to make authenticated API requests.
 
+## Troubleshooting docs for workspaces
```
Comment (nguyen102) [inline solutions/swb-reference/README.md:114]:
```suggestion
## Troubleshooting Guides
```

File: solutions/swb-reference/README.md [hunk 4]
```diff
@@ -111,6 +111,10 @@ Go to `solutions/swb-app` to update `staticRouteConfig.ts` and `staticPermission
...
+
+Workflows to troubleshoot [workspace lifecycle events](docs/troubleshooting-lifecycle-events.md).
```
Comment (nguyen102) [inline solutions/swb-reference/README.md:116]:
```suggestion
[Workspace Lifecycle Troubleshooting Guide](docs/WorkspaceLifecycleTroubleshootingGuide.md).
```

Please rename the markdown to this `WorkspaceLifecycleTroubleshootingGuide.md`

File: solutions/swb-reference/docs/troubleshooting-lifecycle-events.md [hunk 5]
```diff
@@ -0,0 +1,30 @@
+# Troubleshooting workspace lifecycle events
```
Comment (nguyen102) [inline solutions/swb-reference/docs/troubleshooting-lifecycle-events.md:1]:
```suggestion
# Workspace Lifecycle Troubleshooting Guide

```

**Review by nguyen102 — APPROVED** [review]

**Pull request merged**